specs/CookieManagement.md

```diff
@@ -0,0 +1,491 @@
+              EncodeQuote(value.get()) + L", " + L"\"Domain\": " + EncodeQuote(domain.get()) +
+              L", " + L"\"Path\": " + EncodeQuote(path.get()) + L", " + L"\"HttpOnly\": " +
+              BoolToString(httpOnly) + L", " + L"\"Secure\": " + BoolToString(secure) + L", " +
+              L"\"SameSite\": " + EncodeQuote(same_site_as_string) + L", " + L"\"Expires\": ";
+    if (expires == -1)
+    {
+        result += L"This is a session cookie.";
+    }
+    else
+    {
+        result += std::to_wstring(expires);
+    }
+
+    return result + L"\"}";
+    //! [CookieObject]
+}
+
+void ScenarioCookieManagement::GetCookiesHelper(std::wstring uri)
+{
+    //! [GetCookies]
+    if (m_cookieManager)
+    {
+        CHECK_FAILURE(m_cookieManager->GetCookies(
+            uri.c_str(),
+            Callback<ICoreWebView2GetCookiesCompletedHandler>(
+                [this, uri](HRESULT error_code, ICoreWebView2CookieList* list) -> HRESULT {
+                    CHECK_FAILURE(error_code);
+
+                    std::wstring result;
+                    UINT cookie_list_size;
+                    CHECK_FAILURE(list->get_Count(&cookie_list_size));
+
+                    if (cookie_list_size == 0)
+                    {
+                        result += L"No cookies found.";
+                    }
+                    else
+                    {
+                        result += std::to_wstring(cookie_list_size) + L" cookie(s) found on " +
+                                  uri + L".";
+                        result += L"\n\n[";
+                        for (int i = 0; i < cookie_list_size; ++i)
+                        {
+                            wil::com_ptr<ICoreWebView2Cookie> cookie;
+                            CHECK_FAILURE(list->GetValueAtIndex(i, &cookie));
+
+                            if (cookie.get())
+                            {
+                                result += CookieToJsonString(cookie.get());
+                                if (i != cookie_list_size - 1)
+                                {
+                                    result += L",\n";
+                                }
+                            }
+                        }
+                        result += L"]";
+                    }
+                    MessageBox(nullptr, result.c_str(), L"GetCookies Result", MB_OK);
+                    return S_OK;
+                })
+                .Get()));
+    }
+    //! [GetCookies]
+}
+```
+
+## .NET and WinRT
+
+```c#
+void CookieManagementCmdsCanExecute(object sender, CanExecuteRoutedEventArgs e)
+{
+    e.CanExecute = webView != null && webView.CoreWebView2 != null && webView.CoreWebView2.CookieManager != null;
+}
+
+void AddOrUpdateCookieCmdExecuted(object target, ExecutedRoutedEventArgs e)
+{
+    CoreWebView2Cookie cookie = webView.CoreWebView2.CookieManager.CreateCookieWithDetails("CookieName", "CookieValue", ".bing.com", "/");

```

Comment:
What's with the prefix '.' on 'bing.com'? Is '.bing.com' different from 'bing.com'? If so, this needs a comment in the API documentation